controller_report_xls/controllers/main.py

```diff
@@ -57,16 +57,20 @@ def is_integer(value):
 def is_formatted_number(value, thousands_sep, decimal_point):
     set_sign = set([thousands_sep, decimal_point, '-'])
     set_val = set(list(value))
+    is_text = False
     if set_val == set():
-        return False
+        is_text = True
     elif any([val.isalpha() or val.isspace() for val in set_val]):
-        return False
+        is_text = True
     elif not all([val.isdigit() for val in set_val - set_sign]):
-        return False
+        is_text = True
     elif value.count('-') > 1:
-        return False
+        is_text = True
     elif '-' in set_val and not value[0] == '-':
-        return False
+        is_text = True
+
+    if is_text:
+        return value
```

Comment:
I think you do not need to check If is_text due to it is a boolean itself just retur the value. If not you will return  None.